src/mktemp/mktemp.rs

```diff
@@ -0,0 +1,174 @@
+    };
+
+    if args.len() == 1 || matches.opt_present("help") {
+        print_help(&opts);
+        return 0;
+    }
+    if matches.opt_present("version") {
+        println!("{} {}", NAME, VERSION);
+        return 0;
+    }
+
+    if 1 < matches.free.len() {
+        crash!(1, "Too many templates");
+    }
+    // <<
+
+    let make_dir = matches.opt_present("directory");
+    let dry_run = matches.opt_present("dry-run");
+    let suffix_opt = matches.opt_str("suffix");
+
+
+    let template = if matches.free.is_empty() {
+        DEFAULT_TEMPLATE
+    } else {
+        &matches.free[0][..]
+    };
+
+    let (prefix, rand, suffix) = match parse_template(template) {
+        Some((p, r, s)) => match suffix_opt {
+            Some(suf) => if s == "" {
+                (p, r, suf)
+            } else {
+                crash!(1, "Template should ends with 'X' when you specify suffix option.")
+            },
+            None => (p, r, s.to_string())
+        },
+        None => ("",0, "".to_string())
+    };
+
+    if rand < 3 {
+        crash!(1, "Too few 'X's in template")
+    }
+
+    if suffix.chars().any(|c| is_separator(c)) {
+        crash!(1, "suffix cannot contain any path separators");
+    }
+
+
+    let tmpdir = match matches.opt_str("tmpdir") {
+        Some(s) => {
+            if PathBuf::from(prefix).is_absolute() {
+                crash!(1, "template must not be an absolute path when tempdir is specified.");
+            }
+            PathBuf::from(s)
+                
+        },
+        None => env::temp_dir()
+    };
+
+    
+    exec(tmpdir, prefix, rand , &suffix, make_dir, dry_run)
+}
+
+fn print_help(opts: &getopts::Options) {
+    let usage = format!(" Create a temporary file or directory, safely, and print its name.
+TEMPLATE must contain at least 3 consecutive 'X's in last component.
+If TEMPLATE is not specified, use {}, and --tmpdir is implied", DEFAULT_TEMPLATE);
+
+    println!("{} {}", NAME, VERSION);
+    println!("SYNOPSIS");
+    println!("  {} [OPTION]... [FILE]", NAME);
+    println!("Usage:");
+    print!("{}", opts.usage(&usage[..]));
+}
+
+fn parse_template(temp :&str) -> Option<(&str, usize, &str)> {
+    let right = match temp.rfind('X') {
+        Some(r) => r+1,
+        None => return None
+    };
+    let left = temp[..right].rfind(|c| c != 'X').map_or(0, |i| i + 1);
+    let prefix = &temp[..left];
+    let rand = right - left;
+    let suffix = &temp[right..];
+    Some((prefix, rand, suffix))
+}
+
+fn exec(tmpdir: PathBuf, prefix: &str, rand: usize, suffix: &str, make_dir: bool, dry_run: bool, ) -> i32 {
+    // TODO: respect make_dir option
+    if make_dir {
+        crash!(1, "Directory optios is not supported yet. Sorry.");
+    }
+
+
+    let tmpfile = NamedTempFileOptions::new()
+        .prefix(prefix)
+        .rand_bytes(rand)
+        .suffix(suffix)
+    // CAUTION: Creating tempfile regardless of dry_run option.
```

Comment:
Yes, I agree with you. I thought you need to access file system eventually to ensure the generated random file name doesn't exist so I used this method for the sake of simplicity. However I found that `mktemp(1)` don't access the file system and don't assure the uniqueness, I'll implement dry-run correctly.